tests/integration/product-type-export-xlsx.spec.js

```diff
@@ -214,279 +41,273 @@ const readXlsx = function readXlsx (filePath) {
     })
 }
 
-const before = function setup () {
+async function before () {
+  sphereClientConfig = {
+    config: await utils.getClientConfig(),
+  }
+
+  client = await utils.getClient()
   OUTPUT_FOLDER = tempfile()
   fs.mkdirSync(OUTPUT_FOLDER)
-  return getSphereClientCredentials(PROJECT_KEY)
-    .then((sphereCredentials) => {
-      const options = {
-        config: sphereCredentials,
-      }
-      sphereClientConfig = options
-      client = new SphereClient(options)
 
-      productTypeExport = new ProductTypeExport({
-        sphereClientConfig: options,
-        config: {
-          outputFolder: OUTPUT_FOLDER,
-          exportFormat: 'xlsx',
-        },
-      })
-      return deleteAll('productTypes', client)
-        .then(() =>
-          Promise.all(mockProductTypes.map(productType =>
-            client.productTypes.create(productType)
-          ))
-        )
-    })
+  productTypeExport = new ProductTypeExport({
+    sphereClientConfig,
+    config: {
+      outputFolder: OUTPUT_FOLDER,
+      exportFormat: 'xlsx',
+    },
+  })
+  return utils.deleteAll('productTypes', client)
+    .return(mockProductTypes)
+    .map(productType => client.productTypes.create(productType))
 }
 
 test(`writeProductTypes to xlsx
   should write product type to attribute mapping xlsx file`, (t) => {
-  t.timeoutAfter(15000) // 15s
-  before().then(() => {
-    const download = tempWrite.sync()
-    const destination = tempWrite.sync()
-    productTypeExport.downloadProductTypes(download)
-      .then(() => productTypeExport.collectAttributes(download))
-      .then(({ attributeNames }) => {
-        productTypeExport.attributeNames = attributeNames
-        return productTypeExport.collectTypesAndAttributes(download)
-      })
-      .then(({ productTypes }) =>
-        productTypeExport.writeProductTypes(productTypes, destination)
-      )
-      .then(() =>
-        readXlsx(destination)
-      )
-      .then((data) => {
-        const attributes = productTypeExport.attributeNames.join(',')
-        const header = data[0].join(',')
-        t.equal(
-          header,
-          `name,key,description,${attributes}`,
-          `exported row ${header} is equal`
+    t.timeoutAfter(15000) // 15s
+    before().then(() => {
```

Comment:
Here too. We can pass an anonymous `async` function to test and declare `await before()`

Reply:
Updated here 0a632e9